### PR TITLE
Add action for opening settings directory in file explorer

### PR DESCRIPTION
**Pull request by e82eric**
Most of the logic is taken from the original PR (https://github.com/microsoft/terminal/pull/15417) and adapted to work with the palette.

## References and Relevant Issues
[#12382](https://github.com/microsoft/terminal/issues/12382)

### CODE REVIEW

**Comment by DHowett** [comment]
Sorry for the delay! We've been understaffed this week 🙂

**Comment by e82eric** [comment]
No worries,  no rush on my end.

File: src/cascadia/TerminalApp/TerminalPage.cpp
Comment (lhecker) [inline src/cascadia/TerminalApp/TerminalPage.cpp:3134]: The `.c_str()` is unnecessary, if not detrimental, because constructing a string-view from a nullptr is UB.

**Review by lhecker — APPROVED** [review]
Apart from the `.c_str()` call the PR seems fine to me. I'll approve in advance because it'll have to wait for a 2nd approval anyway.

However, I'm not quite sure if we should introduce the 2 additional options just yet.

File: src/cascadia/TerminalApp/TerminalPage.cpp
Comment (e82eric) [reply src/cascadia/TerminalApp/TerminalPage.cpp]: Updated.  (Sorry,  I should have thought that through)

Let me know if I should remove the other 2 actions.  Fwiw, the send input one is what I have found most useful, I don't think I have ever actually used the other 2 outside of testing.  But that is probably just because of my personal workflows.

File: src/cascadia/TerminalSettingsModel/TerminalSettingsSerializationHelpers.h
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsModel/TerminalSettingsSerializationHelpers.h:444]: Discussed this as a team. I see value in opening the directory, but `sendInput` and `clipboard` might be too niche. We'd prefer that you back out those two values please.

File: src/cascadia/TerminalSettingsModel/TerminalSettingsSerializationHelpers.h
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsModel/TerminalSettingsSerializationHelpers.h:442]:
```suggestion
        pair_type{ "directory", ValueType::Directory },
```
I think this would be cleaner

File: src/cascadia/TerminalSettingsModel/defaults.json
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsModel/defaults.json:436]:
```suggestion
        { "command": { "action": "openSettings", "target": "directory" }, "id": "Terminal.OpenSettingsDirectory" },
```

File: src/cascadia/TerminalSettingsModel/Resources/en-US/Resources.resw
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsModel/Resources/en-US/Resources.resw:335]:
```suggestion
    <value>Open the settings file directory</value>
```

**Review by carlos-zamora — CHANGES_REQUESTED** [review]
Thanks for being so patient with us. Sorry it's taken us a long time to get to it.

Got a chance to discuss this with the team. We're interested in being able to open the settings directory, but not the `sendInput` or `clipboard` ones as they may be too niche.

**Comment by e82eric** [comment]
> Thanks for being so patient with us. Sorry it's taken us a long time to get to it.
> 
> Got a chance to discuss this with the team. We're interested in being able to open the settings directory, but not the `sendInput` or `clipboard` ones as they may be too niche.

Sounds good! updated.

**Review by carlos-zamora — APPROVED** [review]
Love it. Thank you!

**Comment by carlos-zamora** [comment]
Gif removed from PR body:

open_settings_in_explorer

**Comment by lhecker** [comment]
Was there a reason why we didn't merge this?

**Review by zadjii-msft — APPROVED** [review]

**Comment by jorgeferreir2k21** [comment]
> Most of the logic is taken from the original PR (https://github.com/microsoft/terminal/pull/15417) and adapted to work with the palette.
> 
> 
> 
> ## References and Relevant Issues
> 
> [#12382](https://github.com/microsoft/terminal/issues/12382)

**Comment by jorgeferreir2k21** [comment]
> Most of the logic is taken from the original PR (https://github.com/microsoft/terminal/pull/15417) and adapted to work with the palette.
> 
> 
> 
> ## References and Relevant Issues
> 
> [#12382](https://github.com/microsoft/terminal/issues/12382)

**Pull request merged**
